### DISCUSSION TITLE
GenerateReleaseNotesConsoleTester.js not running

### DISCUSSION THREAD

**Discussion by jjcpa** [rfennell/AzurePipelines#1110] [Q&A] [answered]
Hi Fennell,  I am quite new with this. I following your instruction to run both npm install npm run build.  I modified release-settings.json.

When I ran 
node GenerateReleaseNotesConsoleTester.js release-settings.json kpjwdlcmkvgrqpeyawvwjxcwfhxp3y6acc2i5xmhuzmujepht123
I got this:
##vso[task.debug]agent.TempDirectory=undefined
##vso[task.debug]agent.workFolder=undefined
##vso[task.debug]loading inputs and endpoints
##vso[task.debug]loaded 0
##vso[task.debug]Agent.ProxyUrl=undefined
##vso[task.debug]Agent.CAInfo=undefined
##vso[task.debug]Agent.ClientCert=undefined
##vso[task.debug]Agent.SkipCertValidation=undefined
Starting Tag XplatGenerateReleaseNotes Local Tester
USAGE: node GenerateReleaseNotesConsoleTester.js --filename settings.json --pat <Azure-DevOps-PAT> --githubpat <Optional GitHub-PAT> --bitbucketuser <Optional Bitbucket User> --bitbucketsecret <Optional Bitbucket App Secret>
Tool exited

Any ideas? 

Thank you very much
Jjcpa

**Comment by rfennell** [comment, marked as answer]
You are not using the command line correctly, you are not entering the parameter names, it should be

```
node GenerateReleaseNotesConsoleTester.js --filename release-settings.json --pat kpjwdlcmkvgrqpeyawvwjxcwfhxp3y6acc2i5xmhuzmujepht123
```

**Reply by jjcpa** [reply]
sorry for the silly errors.  That step is working but I got another error:

oading settings from release-settings.json
Creating Azure DevOps API connections for https://dev.azure.com/Vancity with 'allowRetries' set to 'true' and 'maxRetries' count to '3'
Creating the credential handler using override PAT (suitable for local testing or if the OAUTH token cannot be used)
Getting the current release details
The most common runtime reason for the task to fail is due API ECONNRESET issues. To avoid this failing the pipeline these will be 
treated as warnings and an attempt to generate any release notes possible
Error: Failed request: (401)
    at RestClient.<anonymous> (C:\Vancity\_temp\Extensions\XplatGenerateReleaseNotes\V3\node_modules\typed-rest-client\RestClient.js:202:31)
    at Generator.next (<anonymous>)
    at fulfilled (C:\Vancity\_temp\Extensions\XplatGenerateReleaseNotes\V3\node_modules\typed-rest-client\RestClient.js:6:58)      
    at processTicksAndRejections (internal/process/task_queues.js:95:5) {
  statusCode: 401
}
Total Builds: [0]

Is that because I cannot connect to our Azure DevOps? I tried to debug with Visual Studio Code but kept on getting "Error: spawn C:\Program Files\nodejs\node.exe ENOENT"

thank you for your help

**Reply by rfennell** [reply]
Mostly likely it is an issue with the PAT or team project name. 401 means it can hit the server but not log in.

**Comment by jjcpa** [comment]
thanks you are correct. The token I used is custom one. I have to create one with full access
